### DISCUSSION TITLE
saving qr as image file

### DISCUSSION THREAD

**Discussion by Odido** [chillerlan/php-qrcode#73] [Q&A] [answered]
sorry for asking, but I can not find out how to save the generated qr code to a file in a defined directory (no need to show it on screen) - without user interaction.

Any help appreciated - thx

**Comment by codemasher** [comment, marked as answer]
Just add the path to the destination file as second parameter to the `render()` method like so:
```php
$qrcode->render($data, '/path/to/file.svg');
```
See also: [Advanced usage](https://github.com/chillerlan/php-qrcode/wiki/Advanced-usage) in the wiki.

**Reply by Odido** [reply]
thx. I was trying that but should have omitted the leading /. Without it is working.

**Comment by codemasher** [comment]
Glad it worked now! The safest bet would probably be using the [`__DIR__`](https://www.php.net/manual/en/language.constants.magic.php) constant and/or [`realpath()`](https://www.php.net/manual/en/function.realpath).

**Comment by maximusmcc** [comment]
Is there a better way to do this? It does not work for me for some reason.

**Reply by codemasher** [reply]
What are you trying to do? I can't help you without code example or any further information.

**Reply by maximusmcc** [reply]
Well I was trying to make the program put the qrcode image into a directory but I gave up, lol. Not any real clear direction on this process. Using a different program now. Thanks! :-)

**Reply by codemasher** [reply]
¯\_(ツ)_/¯

**Reply by sabby1977** [reply]
what program you used?

**Reply by maximusmcc** [reply]
I ended up using an older program https://phpqrcode.sourceforge.net/ It works, I didn't need anything fancy like a logo in it or anything but it works! lol
